### PR TITLE
PRODTASKS-615 Pass in product name/version rather than CPE. Improve n…

### PR DESCRIPTION
**Pull request by rnc**
…ame asssignment for binary scans. Improve logging. Fix strange error in email handling. Improve error handling for parse problems.

### CODE REVIEW

File: src/main/java/com/redhat/engineering/srcclr/SrcClrWrapper.java [hunk 1]
```diff
@@ -65,13 +63,16 @@
                     description = "Threshold on which exception is thrown. Only used with CVSS Processor")
     private int threshold = 0;
 
-    @Option( names = { "-p", "--processor" }, defaultValue = "cvss", converter = ProcessorConvertor.class,
+    @Option( names = { "--processor" }, defaultValue = "cvss", converter = ProcessorConvertor.class,
                     description = "Processor (cve|cvss) to use to analyse SourceClear results. Default is cvss")
     private ScanResult processor;
 
-    @Option( names = { "-c", "--cpe" }, defaultValue="", description = "CPE (Product) Name")
+    @Option( names = { "-p", "--product" }, /*required = true,*/ description = "Product Name (in same format as CPE Product Name)")
```
[the remark this answers is not in the document]
Comment (rnc) [reply src/main/java/com/redhat/engineering/srcclr/SrcClrWrapper.java]: Heh ;-) Essentially I planned to make product name mandatory and default the processor to cve but decided to do that in a separate PR.

File: src/main/java/com/redhat/engineering/srcclr/Binary.java [hunk 2]
```diff
@@ -85,17 +82,29 @@ public Void call() throws Exception
         try
         {
             URL processedUrl = new URL( url );
-            name = FilenameUtils.getName( processedUrl.getPath() );
+            filename = FilenameUtils.getName( processedUrl.getPath() );
+            String name = filename;
             File target = new File( urlDownloadLocation.toFile(), name );
 
+            // If the subpackage is NOT a duplicate prefix of the filename then add that as well.
```
[the remark this answers is not in the document]
Comment (rnc) [reply src/main/java/com/redhat/engineering/srcclr/Binary.java]: Please see the long discussion on the ticket and the reason I tried this.

[the remark this answers is not in the document]
Comment (jeremychoi) [reply src/main/java/com/redhat/engineering/srcclr/Binary.java]: Now I got your point by seeing your comment which I missed somehow.

**Comment by jeremychoi** [comment]
LGTM

**Pull request merged**